src/data_gradients/datasets/FolderProcessor.py

```diff
@@ -103,66 +112,15 @@ def _match_file_names(self, all_images_file_names: List[str], all_labels_file_na
 
         return [(image_file_base_names[name], label_file_base_names[name]) for name in common_base_names]
 
-    def is_image(self, filename: str) -> bool:
-        """Check if the given file name refers to image."""
-        return filename.split(".")[-1].lower() in self.image_extensions
-
-    def is_label(self, filename: str) -> bool:
-        """Check if the given file name refers to image."""
-        return filename.split(".")[-1].lower() in self.label_extensions
-
-    @staticmethod
-    def get_filename(file_name: str) -> str:
-        return os.path.splitext(os.path.basename(file_name))[0]
-
-    def __len__(self):
-        return len(self.images_with_labels_files)
-
-    def __getitem__(self, index):
-        return self.images_with_labels_files[index]
-
-    def __iter__(self):
-        for image_label_file in self.images_with_labels_files:
-            yield image_label_file
-
-
-class ImageLabelConfigIterator(ImageLabelFilesIterator):
-    """Iterate over all image and label files in the provided directories."""
-
-    def __init__(
-        self,
-        images_dir: str,
-        labels_dir: str,
-        config_path: str,
-        label_extensions: Sequence[str],
-        image_extensions: Sequence[str] = DEFAULT_IMG_EXTENSIONS,
-        verbose: bool = True,
-    ):
-        """
-        :param images_dir:          The directory containing the images.
-        :param labels_dir:          The directory containing the labels.
-        :param config_path:         Path to the config file. This config file should contain the list of file ids to include.
-        :param label_extensions:    The extensions of the labels. Only the labels with this extensions will be considered.
-        :param image_extensions:    The extensions of the images. Only the images with this extensions will be considered.
-        :param verbose:             Whether to print extra messages.
-        """
-        self.config_path = config_path
-        super().__init__(
-            images_dir=images_dir,
-            labels_dir=labels_dir,
-            label_extensions=label_extensions,
-            image_extensions=image_extensions,
-            verbose=verbose,
-        )
-
-    def get_image_and_label_file_names(self, images_dir: str, labels_dir: str) -> List[Tuple[str, str]]:
-        """Gather all image and label files that are both listed in the config_path and in the directories.
-        :param images_dir:      The directory containing the images.
-        :param labels_dir:      The directory containing the labels.
-        :return:                A list of tuple(<path-to-image>, <path-to-label>).
```

Comment:
I realized that in lots of cases I wrote the following code:
```python
        if config_path is None:
            self.image_label_tuples = ImageLabelFilesIterator(
                images_dir=os.path.join(root_dir, images_subdir),
                labels_dir=os.path.join(root_dir, labels_subdir),
                image_extensions=image_extensions,
                label_extensions=label_extensions,
                verbose=verbose,
            )
        else:
            self.image_label_tuples = ImageLabelConfigIterator(
                images_dir=os.path.join(root_dir, images_subdir),
                labels_dir=os.path.join(root_dir, labels_subdir),
                config_path=os.path.join(root_dir, config_path),
                image_extensions=image_extensions,
                label_extensions=label_extensions,
                verbose=verbose,
            )
```
This made me realize that we should not have two classes for this, but just one.
The downside is that this class is starting to be relatively big which is never good, but I think it is reasonable